the first chunk fixes build errors,
the non_stop part below fixes thread debugging being available
by default without setting any breakpoints or usage of libthread_db.

--- gdb-7.4.org/gdb/linux-nat.c	2013-08-10 05:24:24.651000003 +0000
+++ gdb-7.4/gdb/linux-nat.c	2013-08-10 05:25:50.966000003 +0000
@@ -71,6 +71,14 @@
 # endif
 #endif /* HAVE_PERSONALITY */
 
+#ifndef __SIGRTMIN
+#define __SIGRTMIN SIGRTMIN
+#endif
+
+#ifndef W_STOPCODE
+#define W_STOPCODE(sig) ((sig) << 8 | 0x7f)
+#endif
+
 /* This comment documents high-level logic of this file.
 
 Waiting for events in sync mode
diff -rup gdb-7.10-old/gdb/mips-linux-nat.c gdb-7.10/gdb/mips-linux-nat.c
--- gdb-7.10-old/gdb/mips-linux-nat.c	2015-09-08 00:15:37.000000000 +0200
+++ gdb-7.10/gdb/mips-linux-nat.c	2015-09-08 00:15:58.000000000 +0200
@@ -31,7 +31,7 @@
 #include "gdb_proc_service.h"
 #include "gregset.h"
 
-#include <sgidefs.h>
+#include <asm/sgidefs.h>
 #include <sys/ptrace.h>
 #include <asm/ptrace.h>
 
